apps/client/src/components/Questions.js

```diff
@@ -114,68 +116,67 @@ const Questions = ({ checker, topic: { slug } }) => {
     checker.next();
   }
 
-  // @TODO: Refactor this map function
-  return checker.stack.filter(uniqueFilter).map((q, i) => {
-    // Define userAnswer
-    let userAnswer;
-    if (q.options) {
-      userAnswer = q.answer;
-    } else {
-      const responseObj = booleanOptions.find((o) => o.value === q.answer);
-      userAnswer = responseObj?.label;
-    }
+  return (
+    checker.stack
+      // Remove duplicate questions (do we really need this?)
+      .filter(uniqueFilter)
+      // Loop through questions
+      .map((q, i) => {
+        // Define userAnswer
+        const booleanAnswers =
+          !q.options && booleanOptions.find((o) => o.value === q.answer);
+        const userAnswer = q.options ? q.answer : booleanAnswers?.label;
 
-    // Show the current active question
-    if (q === checker.stack[questionIndex] && !finishedQuestions) {
-      return (
-        <StepByStepItem
-          active
-          highlightActive
-          customSize
-          heading={q.text}
-          key={`question-${q.id}-${i}`}
-          // Overwrite the line between the Items
-          style={{ marginTop: -1, borderColor: "white" }}
-        >
-          <Question
-            question={q}
-            questionIndex={questionIndex}
-            checker={checker}
-            onSubmit={onQuestionNext}
-            setEditQuestion={setEditQuestion}
-            editQuestion={editQuestion}
-            onGoToPrev={onQuestionPrev}
-            userAnswer={userAnswer}
-            showNext
-          />
-        </StepByStepItem>
-      );
-    } else if (userAnswer) {
-      // Show answered questions (beware: boolean `false` is a possible answer)
-      return (
-        <StepByStepItem
-          checked
-          customSize
-          heading={q.text}
-          key={`question-${q.id}-${i}`}
-        >
-          <Paragraph gutterBottom={0}>
-            {removeQuotes(userAnswer)}
-
-            <Button
-              style={{ marginLeft: 20 }}
-              onClick={() => onGoToQuestion(i)}
-              variant="textButton"
-            >
-              Wijzig
-            </Button>
-          </Paragraph>
-        </StepByStepItem>
-      );
-    }
-    // Hide unanswered questions (eg: on browser refresh)
-    return null;
-  });
+        // Define if question is the current one
+        const isCurrentQuestion =
+          q === checker.stack[questionIndex] &&
+          // isActive("questions"); @TODO needs to be added (in branch feature/refactor-stepper-logic)
+          !finishedQuestions; // @TODO needs to be removed (in branch feature/refactor-stepper-logic)
+
+        // Hide unanswered questions (eg: on browser refresh)
+        if (!isCurrentQuestion && !userAnswer) return null;
```

Comment:
We might not even need this, but lets keep it as fallback

Reply:
I've double-checked it and we need it, because otherwise we get an error on refresh